docs/content/developer/workshops/core/II-shared-objects-dynamic-fields-ptbs.mdx

```diff
@@ -0,0 +1,331 @@
+---
+title: Workshop II Practical — Shared State, Indexing & PTBs
+description: Create a shared Registry, index student-to-grant IDs, and compose an atomic Programmable Transaction Block (PTB) that mints and indexes in one go.
+tags:
+  - tutorial
+  - move-sc
+  - typescript
+---
+
+# Workshop II Practical — Shared State, Indexing & PTBs
+
+## Quick recap
+
+- We published a Move package and received an `AdminCap` in `init` (publisher‑owned).
+- We minted a `Grant` object for a student and emitted a `GrantMinted` event.
+- We verified the new owned object on-chain.
+
+## What’s next
+
+We need a way to look up a student’s current grant. This is important for portals and verifiers to render status without replaying history.
+
+- Missing piece: a shared, canonical index mapping `student -> grant_id` and a way to update it whenever we mint.
+- Why it matters:
+  - Portals and verifiers need O(1)‑style lookups to render “current grant” without replaying history.
+  - Multiple apps can coordinate on one source of truth instead of bespoke caches.
+  - A PTB lets us mint and index atomically — either both succeed or neither does — reducing partial failures and data drift.
+
+What you’ll build next:
+
+- Share an on-chain `Registry` object that anyone can read
+- Add a write-gated `index_grant` entry for admins to upsert student → grant
```

Comment:
Maybe an alternative to upsert might make more sense here 🤔 Might be "to complicated" right at the start